posydon/utils/posydonerror.py

```diff
@@ -0,0 +1,71 @@
+"""The POSYDON exception class and subclasses for more specific errors."""
+
+import copy
+from posydon.binary_evol.binarystar import BinaryStar
+from posydon.binary_evol.singlestar import SingleStar
+
+
+class POSYDONError(Exception):
+    """General POSYDON exception class."""
+
+    def __init__(self, message="", objects=None):
+        """General POSYDON exception.
+
+        Parameters
+        ----------
+        message : str
+            POSYDONError message.
+        objects : None or list of objects.
+            A list of accompanied objects (or None if not set), that will be
+            handled differently when `str` method is used.
+
+        """
+        self.message = message
+        # copy the objects: we must know their state at the moment of the error
+        self.objects = copy.copy(objects)
+        super().__init__(self.message)
+
+    def __str__(self):
+        """Create the text that accompanies this exception."""
+        result = ""
+        if self.objects is not None:
+            if isinstance(self.objects,list):
```

Comment:
This might not work in the case the object is a real list itself.

Reply:
We have fixed this.